code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -113,6 +113,92 @@
 		return
 	return held_item.GetID()
 
+/**
+ * Used to fetch all the cash a vendor can access from the human
+ * See /mob/living/carbon/human/proc/get_cash_list() proc for the list of items and inventory slots it searches through
+ */
+/mob/living/carbon/human/proc/get_accessible_cash()
+	var/available_cash = 0
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list))
+		return 0
+	for(var/found_item in cash_list)
+		if(istype(found_item, /obj/item/holochip))
+			var/obj/item/holochip/chip_stack = found_item
+			available_cash += chip_stack.credits
+		if(istype(found_item, /obj/item/card/id))
+			var/obj/item/card/id/id_card = found_item
+			available_cash += id_card.registered_account.account_balance
+	return available_cash
+
+/**
+ * Used by vendors to see if the human can spend a certain amount of cash.
+ * Returns FALSE if they cannot, TRUE if they can and withdraws the cash.
+ * Arguments:
+ * * to_spend = how much cash needs to be deducted
+ */
+/mob/living/carbon/human/proc/spend_cash(to_spend)
+	if(!to_spend)
+		return FALSE
+	if(to_spend > get_accessible_cash()) //If we don't have enough money, early return
+		return FALSE
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list)) //Another check, just in case
+		return FALSE
```

Comment:
```suggestion
	var/list/cash_list = get_cash_list()
	if(!length(cash_list))
		return FALSE
	if(to_spend > get_accessible_cash(cash_list)) //If we don't have enough money, early return
		return FALSE
```
with this suggestion, change `/mob/living/carbon/human/proc/get_accessible_cash()` to override `cash_list` list inside like how `get_cash_list` does. (it's because you calculate get_cash_list() twice but if you did it already, you don't have to do it)

Reply:
Implemented this in commit [0e41611](https://github.com/BeeStation/BeeStation-Hornet/pull/9310/commits/0e416116f6810a22473649a5a17189e2f07058db)